### ISSUE TITLE
Process inline stem macros inside section titles

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor-mathematical#21]
Process inline stem macros inside the text of a section title. This should be pretty much the same logic as processing prose blocks, only the source will need to be retrieved from and assigned to the title instance variable. After assigning the title instance variable, be sure to clear the subbed_title instance variable.

**Comment by ProgramFan** [comment]
Inline stems in section title are handled by the above commit. However, the image does not scale with the line height. Are there any clues on how can I scale the equation height w.r.t. the line height? Both in html and in pdf. @mojavelinux

**Comment by ProgramFan** [comment]
It seems I can use explicit css style `style="height: 1em"` to achieve that in the html backend, but then I hard coded something special for html. For the PDF, no proper ways seem possible as described in the document on how inline images are scaled. @mojavelinux

**Comment by ProgramFan** [comment]
Lack of this makes inline stems in section titles look really bad, as the produced width and height are for normal paragraphs. How can I work around this? @mojavelinux

**Comment by tstumm** [comment]
I think Mathematical supports the usage of \large \Large \huge \Huge and so on. Injecting this would be suitable workaround. A option for the font size in Mathematical itself would be better though.

**Comment by ProgramFan** [comment]
Nice idea, @tstumm. Then users can workaround this. Since not so many documents use inline stems in section titles, I would rather not injecting those macros into the formulee theirselves.

**Comment by resort-diaper** [comment]
When trying to convert a file with inline stem macros inside a section heading the following error is raised:
```
asciidoctor: FAILED: simple.adoc: Failed to load AsciiDoc document - instance variable @subbed_title not defined
``` 

Here is a simple test document:
```
= Simple Document

== Introduction latexmath:[x]

Simple test Asciidoctor.
```

The issue appears to be because Asciidoctor no longer contains the `subbed_title` instance variable; it was removed in this commit: https://github.com/asciidoctor/asciidoctor/commit/cf5f57bc241ee6e51bf9d893f40b1a2128793d70

**Comment by mojavelinux** [comment]
This extension should not be accessing this instance variable directly. Instead, the following code:

```
    if source_modified
      sect.instance_variable_set :@title, text
      sect.remove_instance_variable :@subbed_title
    end
```

should be written as:

```
    if source_modified
      sect.title = text
    end
```

That automatically flushes the cached converted title and the title will be converted again on next access.

**Comment by mojavelinux** [comment]
This issue was resolves by #21. See c0f8c4c9711235bafb10b8380e101f489d0e5679

@ProgramFan Please close issues once they have been fixed. Thanks!

I have included the fix for the problem I mentioned in my last comment in #93. The code in question is not compatible with Asciidoctor 2, so the proposed change is required for it to function correctly.
